### PR TITLE
TableGC: fix PURGE race condition

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

See https://github.com/vitessio/vitess/issues/12504 for problem description

This PR fixes the race condition by eliminating the execution of one code path as follows:

If `PURGE` isn't a handled state (e.g. running on MySQL `8.0.23` and above), then we now never populate the in-memory `purgingTables` map. It's left empty at all times.

As such, the goroutine that runs on `purgeRequestsChan` has nothing to do, as it attempts to pick a table out of said in-memoty map, which is empty. So that code path is eliminated.

`checkTables()` is now the only owner for any existing `_vt_PURGE_*` tables, should these ever exist. It's fine if they do exist, and `checkTables()` will transition them into the next state.

Speaking of which, the fact the Online DDL Executor used to explicitly rename tables into `_vt_PURGE_*` is strictly speaking OK, not a bug, but then again it's also wasteful. going forward, we expect `PURGE` state to go extinct (as we disable it by default in `8.0.23`). So why bother creating a table in `PURGE` state.

So in this PR we also change the Online DDL Executor behavior: instead of renaming an artifact table into a `_vt_PURGE...`, it renames it into a `_vt_HOLD...` with a timestamp that is in the past. This way, next iteration of TableGC will transition the table into next state. To clarify, `_vt_PURGE` is also fine, and Table GC would transition that to the next state, in very much the same way. It's just that it's surprising to see a `_vt_PURGE` table when `PURGE` is not a supported  state. I think it's now cleaner.

Lastly, added an `Info` log entry upon `Open()` that clarifies the MySQL version and the supported states.

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/12504

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by shlomi-noach** [comment]
All `Upgrade Downgrade` tests failing because of `golang` version. To be ignored.

**Review by dbussink — APPROVED** [review]

File: go/vt/vttablet/onlineddl/executor.go [hunk 1]
```diff
@@ -3799,17 +3799,21 @@ func (e *Executor) deleteVReplicationEntry(ctx context.Context, uuid string) err
 }
 
 // gcArtifactTable garbage-collects a single table
-func (e *Executor) gcArtifactTable(ctx context.Context, artifactTable, uuid string, t time.Time) (string, error) {
+func (e *Executor) gcArtifactTable(ctx context.Context, artifactTable, uuid string, timestampInThePast time.Time) (string, error) {
```
Comment (deepthi) [inline go/vt/vttablet/onlineddl/executor.go:3802]: nit: it seems weird to rename this function parameter. Those are the semantics the caller is using, this function simply generates the rename statement with whatever time is given to it.

**Review by deepthi — APPROVED** [review]
Just one comment, the rest LGTM

File: go/vt/vttablet/onlineddl/executor.go (hunk 1 above)
Comment (shlomi-noach) [reply go/vt/vttablet/onlineddl/executor.go:3802]: renamed back to original generic name `t`

**Pull request merged**